src/ciadmin/generate/worker_pools.py

```diff
@@ -27,11 +27,27 @@ def is_invalid_aws_instance_type(invalid_instances, zone, instance_type):
 
 
 def is_invalid_gcp_instance_type(invalid_instances, zone, machine_type):
-    family, _ = machine_type.split("-", 1)
-    return any(
-        (zone in entry["zones"]) and (family in entry["families"])
-        for entry in invalid_instances
-    )
+    # Parse machine type: family-configuration-cpus[-suffix]
+    # e.g., "c4d-standard-8-lssd" -> family="c4d", suffix="lssd"
+    parts = machine_type.split("-")
+    family = parts[0]
+    # If the last part is not numeric, it's a suffix
+    suffix = parts[-1] if not parts[-1].isdigit() else None
```

Comment:
It looks like GCP instances are always in the form of: $family-$subtype-$cpus-$suffix (the suffix seems like it's sometimes disk related, sometimes `metal`, and possibly something else).

Assuming this is true, this is safer to parse by splitting on `-` and checking how many parts you have. (3 parts would be no suffix, 4 parts would have a suffix.)